src/tests.rs

```diff
@@ -4334,6 +4334,164 @@ export function foo(bar: any): asserts bar {}
     }
   ]);
 
+  json_test!(infer_object_literal,
+    r#"export const a = {
+      a: "a",
+      b: new Map<string, number>(),
+      c: { d: "d" },
+      d(e: string): void {},
+      f: (g: string): void => {},
+      get h(): string {
+        return "h";
+      },
+      set h(value: string) {
+    
+      },
+    };"#;
+    [
+      {
+        "kind": "variable",
+        "name": "a",
+        "location": {
+          "filename": "file:///test.ts",
+          "line": 1,
+          "col": 0
+        },
+        "declarationKind": "export",
+        "variableDef": {
+          "tsType": {
+            "repr": "",
+            "kind": "typeLiteral",
+            "typeLiteral": {
+              "methods": [],
+              "properties": [
```

Comment:
The first challenge is that this appears to not be deterministic, but other situations are.  Having it not be deterministic I think is going to cause problems.  Like when we generate anchors, we would expect it to be consistent ordering, which when declared it obviously has consistent ordering.  I am not sure offhand why it is not deterministic here.

Reply:
Used IndexMap instead of HashMap.